ast2vec/model2/proxbase.py

```diff
@@ -0,0 +1,155 @@
+from collections import defaultdict, deque
+from itertools import permutations, product
+import os
+from typing import Dict, Tuple
+
+import numpy
+from scipy.sparse import coo_matrix, diags
+
+from ast2vec.coocc import Cooccurrences
+from ast2vec.token_parser import TokenParser
+from ast2vec.uast import UASTModel
+from ast2vec.model2.base import Model2Base
+
+EDGE_TYPES = ["r", "t", "rt", "R", "T", "RT"]
+
+
+class ProxBase(Model2Base):
+    """
+    Contains common utilities for proximity matrix models.
```

Comment:
Can you please write a few sentences what is a proximity matrix, what is the algorithm and so on.